app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,142 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    private val homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+    private val popularCategories = listOf(
+        MovieCategory.POPULAR_STREAMING,
+        MovieCategory.POPULAR_ON_TV,
+        MovieCategory.POPULAR_FOR_RENT,
+        MovieCategory.POPULAR_IN_THEATRES,
+    )
+
+    private val nowPlayingCategories = listOf(
+        MovieCategory.PLAYING_MOVIES,
+        MovieCategory.PLAYING_TV,
+    )
+
+    private val upcomingCategories = listOf(
+        MovieCategory.UPCOMING_TODAY,
+        MovieCategory.UPCOMING_THIS_WEEK,
+    )
+
+    private val popularCategorySelected = MutableStateFlow(popularCategories[0])
+    private val nowPlayingCategorySelected = MutableStateFlow(nowPlayingCategories[0])
+    private val upcomingCategorySelected = MutableStateFlow(upcomingCategories[0])
+
+
+    private val popularViewStateInternal: MutableStateFlow<HomeMovieCategoryViewState> =
+        MutableStateFlow(HomeMovieCategoryViewState())
+    val popularViewState = popularViewStateInternal.asStateFlow()
+
+    private val nowPlayingViewStateInternal: MutableStateFlow<HomeMovieCategoryViewState> =
+        MutableStateFlow(HomeMovieCategoryViewState())
+    val nowPlayingViewState = nowPlayingViewStateInternal.asStateFlow()
+
+    private val upcomingViewStateInternal: MutableStateFlow<HomeMovieCategoryViewState> =
+        MutableStateFlow(
+            HomeMovieCategoryViewState()
+        )
+    val upcomingViewState = upcomingViewStateInternal.asStateFlow()
+
+
+    init {
+        viewModelScope.launch {
+            movieRepository.popularMovies(popularCategorySelected.value).collect { movies ->
+                popularViewStateInternal.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = popularCategories,
+                    movies = movies,
+                    selectedMovieCategory = popularCategorySelected.value
+                )
+            }
+        }
```

Comment:
The problem here is that you are not observing popularSelected (same problem with upcoming and nowPlaying), so it won't be updated when flow emits new value (in your case selected tab is updated because you are changing the entire viewModel, which is not somethign we want to do) . 
So you would have to do something like this:
```
movieRepository.popMovies().collect { movies -> 
   popularSelected.collect { selected -> 
     ....
     }
     }
```
But you won't do this since we don't want to collect flow just to change value. We don't have to collect flow until we wish to do something with it (this will usually be in Composable function), we just have to change it and we have plenty of extension functions which can do this for us
So you will have to do something like this:
``` selected.flatMapLatest { repo.movies().map{ ...}}.stateIn(...)```